sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -714,13 +719,18 @@ public void processElement(ProcessContext c) throws Exception {
                   }
                 });
         ++recordsWritten;
+        seenWindows.add(window);
       }
 
       @FinishBundle
-      public void finishBundle() throws Exception {
+      public void finishBundle(FinishBundleContext c) throws Exception {
         bigtableWriter.flush();
         checkForFailures();
         LOG.debug("Wrote {} records", recordsWritten);
+
+        for (BoundedWindow window : seenWindows) {
+          c.output(BigtableWriteResult.create(recordsWritten), window.maxTimestamp(), window);
```

Comment:
I think it's better to aggregate records per window and output the number of records written for a given window. Otherwise if a following step aggregate values across windows they will see a total more than the total number of records written. WDYT ? Either way we should clearly document what we are outputting here.

Reply:
I think we kind of talked about this before.  If someone wanted that metric, they could do so using a Sum on the output of this.  I don't see a good reason to do that in here, doing so would require a GBK on the output which I don't think this transform should force onto people.

Reply:
I agree with @chamikaramj here: as-is, the returned value is kinda useless because it can't be correctly aggregated - if we see 3 records in W1 and 5 records in W2 in a bundle, the current code will output (8, W1) and (8, W2) with no way for a follow-up Sum to untangle them back. You don't need a GBK, you just need a `HashMap<BoundedWindow, Integer>`.

Reply:
I'd rather just go back to outputting null at this point...if you used a map of window -> counts you'd need to then figure out when the correct time to expire the items from the map was as well.

Reply:
Not sure what's the issue with expiration here? Just clear the map at the beginning of the bundle; you get 1 result per window per bundle.

However if you feel doubtful, then I'm also fine with removing BigtableWriteResult and getting back to outputting null, but please make it a `PCollection<?>` rather than `PCollection<Void>` so that later we can reintroduce BigtableWriteResult without breaking compatibility.